### PR TITLE
literal workaround

### PR DESCRIPTION
**Pull request by ahuang11**
Since quotations `'"table"."name"'` is unconventional, the LLMs would hallucinate table names, i.e. drop quotations, and it would crash the Pydantic model, raising an invalid error. So, to workaround that, replaces everything except alphanumeric with `_` which the LLM understands/copies well, then remaps it back to the original table name with all the special characters after the LLM chooses the table.

### CODE REVIEW

File: lumen/ai/agents.py [hunk 1]
```diff
@@ -460,6 +460,8 @@ async def _select_relevant_table(self, messages: list[Message]) -> tuple[str, Ba
         available_sources = self._memory["available_sources"]
         tables_to_source, tables_schema_str = await gather_table_sources(available_sources)
         tables = tuple(tables_to_source)
+
+        table = None
```
Comment (ahuang11) [inline lumen/ai/agents.py:464]:
```suggestion
```

File: lumen/ai/agents.py [hunk 2]
```diff
@@ -474,15 +476,19 @@ async def _select_relevant_table(self, messages: list[Message]) -> tuple[str, Ba
                     elif len(tables) > FUZZY_TABLE_LENGTH:
                         tables = await self._get_closest_tables(messages, tables)
                     system_prompt = self._render_prompt("select_table", tables_schema_str=tables_schema_str)
-                    table_model = make_table_model(tables)
+                    tables_aliases = {
+                        re.sub(r'[^a-zA-Z0-9_]', '_', table): table for table in tables
+                    }
+                    print(tables_aliases, list(tables_aliases), "TABLES...")
```
Comment (ahuang11) [inline lumen/ai/agents.py:482]:
```suggestion
```

File: lumen/ai/coordinator.py [hunk 3]
```diff
@@ -655,7 +657,10 @@ async def _compute_execution_graph(self, messages: list[Message], agents: dict[s
             for table in src.get_tables():
                 tables[table] = src
 
-        reason_model, plan_model = make_plan_models(agent_names, list(tables))
+        tables_aliases = {
+            re.sub(r'[^a-zA-Z0-9_]', '_', table): table for table in tables
+        }
+        reason_model, plan_model = make_plan_models(agent_names, list(tables_aliases))
```
Comment (ahuang11) [inline lumen/ai/coordinator.py:663]:
```suggestion
        reason_model, plan_model = make_plan_models(agent_names, tuple(tables_aliases))
```

File: lumen/ai/agents.py [hunk 4]
```diff
@@ -474,15 +476,19 @@ async def _select_relevant_table(self, messages: list[Message]) -> tuple[str, Ba
...
+                    table_model = make_table_model(list(tables_aliases))
```
Comment (ahuang11) [inline lumen/ai/agents.py:483]:
```suggestion
                    table_model = make_table_model(tuple(tables_aliases))
```

**Review by philippjfr — COMMENTED** [review]
Honestly, I believe you this works but it seems pretty hacky.

**Comment by ahuang11** [comment]
Superseded by https://github.com/holoviz/lumen/pull/799

**Pull request closed without merging**